Mathlib/RepresentationTheory/GroupCohomology/Functoriality.lean

```diff
@@ -306,6 +321,87 @@ lemma map_comp_isoH1_hom :
+    rintro g x hx ⟨s, hs⟩
+    simpa using congr(A.ρ g $(hx ⟨(g⁻¹ * s * g), Subgroup.Normal.conj_mem' ‹_› s hs g⟩))
+
+/-- The inflation map `H¹(G ⧸ S, A^S) ⟶ H¹(G, A)` is a monomorphism. -/
+instance : Mono (H1InfRes A S).f := by
+  rw [ModuleCat.mono_iff_injective, injective_iff_map_eq_zero]
+  intro x hx
+  induction x using H1_induction_on
+  simp_all only [H1InfRes_X₂, H1InfRes_X₁, H1InfRes_f, H1π_comp_H1Map_apply (QuotientGroup.mk' S),
+    Submodule.Quotient.mk_eq_zero]
+  rcases hx with ⟨y, hy⟩
+  refine ⟨⟨y, fun s => ?_⟩, Subtype.ext <| funext fun g => Quotient.inductionOn' g
+    fun g => Subtype.ext <| congr_fun (Subtype.ext_iff.1 hy) g⟩
+  simpa [sub_eq_zero, shortComplexH1, moduleCatToCycles, (QuotientGroup.eq_one_iff s.1).2 s.2]
+    using congr_fun (Subtype.ext_iff.1 hy) s.1
+
+/-- Given a `G`-representation `A` and a normal subgroup `S ≤ G`, the short complex
+`H¹(G ⧸ S, A^S) ⟶ H¹(G, A) ⟶ H¹(S, A)` is exact. -/
+lemma H1InfRes_exact : (H1InfRes A S).Exact := by
+  rw [moduleCat_exact_iff_ker_sub_range]
+  intro x hx
+  induction x using H1_induction_on with | @h x =>
+  simp_all only [H1InfRes_X₂, H1InfRes_X₃, H1InfRes_g, H1InfRes_X₁, LinearMap.mem_ker,
+    H1π_comp_H1Map_apply S.subtype, Submodule.Quotient.mk_eq_zero, H1InfRes_f]
+  rcases hx with ⟨(y : A), hy⟩
+  have h1 := (mem_oneCocycles_iff x).1 x.2
+  have h2 : ∀ s ∈ S, x s = A.ρ s y - y :=
+    fun s hs => (groupCohomology.oneCocycles_ext_iff.1 hy ⟨s, hs⟩).symm
+  refine ⟨H1π _ ⟨fun g => Quotient.liftOn' g (fun g => ⟨x.1 g - A.ρ g y + y, ?_⟩) ?_, ?_⟩, ?_⟩
+  · intro s
+    calc
+      _ = x (s * g) - x s - A.ρ s (A.ρ g y) + (x s + y) := by
+        simp [add_eq_of_eq_sub (h2 s s.2), sub_eq_of_eq_add (h1 s g)]
+      _ = x (g * (g⁻¹ * s * g)) - A.ρ g (A.ρ (g⁻¹ * s * g) y - y) - A.ρ g y + y := by
+        simp only [mul_assoc, mul_inv_cancel_left, map_mul, LinearMap.mul_apply, map_sub,
+          Representation.self_inv_apply]
+        abel
+      _ = x g - A.ρ g y + y := by
+        simp [eq_sub_of_add_eq' (h1 g (g⁻¹ * s * g)).symm,
+          h2 (g⁻¹ * s * g) (Subgroup.Normal.conj_mem' ‹_› _ s.2 _)]
+  · intro g h hgh
+    have := congr(A.ρ g $(h2 (g⁻¹ * h) <| QuotientGroup.leftRel_apply.1 hgh))
+    simp_all [← sub_eq_add_neg, sub_eq_sub_iff_sub_eq_sub]
+  · rw [mem_oneCocycles_iff]
+    intro g h
+    induction g using QuotientGroup.induction_on with | @H g =>
+    induction h using QuotientGroup.induction_on with | @H h =>
+    apply Subtype.ext
+    simp [← QuotientGroup.mk_mul, h1 g h, sub_add_eq_add_sub, add_assoc]
+  · symm
+    simp only [ModuleCat.hom_ofHom, oneCocycles.val_eq_coe, Submodule.mkQ_apply,
+      H1π_comp_H1Map_apply, Submodule.Quotient.eq]
+    use y
+    refine Subtype.ext <| funext fun g => ?_
+    simp only [moduleCatToCycles_apply_coe, AddSubgroupClass.coe_sub]
+    simp [shortComplexH1, mapOneCocycles_comp_subtype_apply (A := A.quotientToInvariants S)
+      (QuotientGroup.mk' S), oneCocycles.coe_mk (A := A.quotientToInvariants S), ← sub_sub]
```

Comment:
I tried to get this down to one simp call and failed -- do you know what's going on?